src/app/home/home.component.ts

```diff
@@ -18,12 +19,18 @@ export class HomeComponent implements OnInit {
   public dataSource: Data[] = [];
   public isLoadingResults = true;
   buttonValue = '';
-  constructor (private weather: WeatherAPIService, private visibility: PageVisibilityService, readonly notificationService: NotificationService, readonly modal$$: ModalService) {
+  constructor (private weather: WeatherAPIService, private visibility: PageVisibilityService, readonly notificationService: NotificationService, readonly modal$$: ModalService, private changeRef: ChangeDetectorRef) {
   }
 
   ngOnInit (): void {
     this.loadData()
-    setTimeout(() => { this.hiddenOptions.push('option 3') }, 1000)
+    setTimeout(() => {
+      this.hiddenOptions.push('option 3')
+      this.changeRef.detectChanges()
+      this.changeRef.markForCheck()
+      // this.hiddenOptions = [...this.hiddenOptions, 'option 3']
+      // this.hiddenOptions = [...this.hiddenOptions] // Had to update reference of type in order to use OnPush
+    }, 5000)
```

Comment:
hm, I guess the most important part that I want you to understand is that:
choosing OnPush strategy means that the component will be as "dumb" as possible and, ideally, of course,
but we should avoid mutating that state ( in your example, state is hiddenOptions )
Ignoring this rule of thumb, you're trapping yourself into the situation that you have right now:
- onpush blocks the ChangeDetection mechanism, but you still need a lot of places to trigger it.

What are the rules to remember ( let's take that [article about OnPush](https://medium.com/angularwave/onpush-your-new-default-ba3fd5bc9f6e#:~:text=In%20OnPush%20there,a%20little%20deeper!) as source) :
- approach of smart and dumb components will help to allocate all api/storage connection and keep representational components simple and with OnPush
- RxJs helps a lot there, because converting 
- avoid subscribe() if possible, letting | async pipe do it's job